scripts/screeps-private-smoke.py

```
@@ -0,0 +1,765 @@
+    )
+
+
+def poll_stats(config: HarnessConfig, seconds: int) -> list[dict[str, Any]]:
+    """Collect private-server /stats samples for the observation window."""
+    observations: list[dict[str, Any]] = []
+    deadline = time.time() + seconds
+    while time.time() <= deadline:
+        try:
+            stats = http_json(f"{config.http_url}/stats")
+            observations.append({k: stats.get(k) for k in STATS_FIELDS})
+        except Exception as exc:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)